src/palace.pyx

```diff
@@ -119,6 +121,16 @@ def query_extension(name: str) -> bool:
     return devmgr.query_extension(name)
 
 
+def current_context() -> Optional[Context]:
+    """The context that is currently used."""
```

Comment:
Oops I missed this, it should be in verb form (return etc.) and the edge case (`nullptr`) is missed.  Also in `__all__`, try joining two of the three lines together (`query_extension` looks a bit lonely, and I'm easy to get emotional).

I refuse to believe that there's no tutorial online on how to split things up, however, to save time, this is what I believe all you have to do is
1. Get all changes to a patch, e.g. `git diff master > patch`
2. Switch to a temporary branch (could be master)
3. Delete your branch
4. Checkout a new branch of the same name
5. Git apply the patch
6. ?
7. Profit!

Reply:
> I refuse to believe that there's no tutorial online on how to split things up.

There are... it's just that I have to do more checking and editing, and I'm lazy.

Reply:
> `Context` docstring seems not updated as well

I'm not very good at documenting so but I think you mean you mean:

```
...
is equivalent to::  # Is this a typo?
    use_context(context)
    try:
        ...
    finally:
        use_context(None)  # THIS
        context.destroy()
...
```

So I am suggesting trying that line into
```
finally:
    # Doing some terribly complex things to retrieve the previous context here
    use_context(previous)
    context.destroy()
```

Reply:
Also thanks for the suggestion, because I've just struggled with splitting one commit into two. But I'm a bit worried by the (?) part.

Reply:
It's the add and committing part.